packages/ld_default.rb

```diff
@@ -21,28 +21,42 @@ def self.build
 
   type=\$(file ld | cut -d' ' -f2 | tr -d '\\n')
 
-  if [ \"\${type}\" == \"ELF\" ]; then
-    current=\$(find . -inum \$(ls -i ld | cut -d' ' -f1) | fgrep 'ld.')
-  elif [ \"${type}\" == \"symbolic\" ]; then	
+  if [[ \"\${type}\" == \"ELF\" ]]; then
+    current=\$(basename $(find . -inum \$(ls -i ld | cut -d' ' -f1) | fgrep 'ld.'))
+  elif [[ \"${type}\" == \"symbolic\" ]]; then	
     current=\$(basename \$(readlink ld))
-  else
+  elif [[ \"${type}\" == \"Bourne-Again\" ]]; then
     current=\$(basename \$(tail -1 ld | cut -d' ' -f1))
+  else # Fall back to \"Unknown\" if linker was not identified
+    current=\"Unknown\"
   fi
 
-  echo
-  echo \"  Current default linker: \${current}\"
-  echo
-  echo '  Enter the new default linker:'
-  echo
-  echo '  b = ld.bfd'
-  echo '  g = ld.gold'
-  echo '  l = ld.lld'
-  echo '  0 = Cancel'
-  echo
+  if [[ -z \"\${1}\" ]]; then
+    echo
+    echo \"  Current default linker: \${current}\"
+    echo
+    echo '  Enter the new default linker:'
+    echo
+    echo '  b = ld.bfd'
+    echo '  g = ld.gold'
+    echo '  l = ld.lld'
+    echo '  0 = Cancel'
+    echo
+  else
+    echo \"\${current}\" | cut -d'.' -f2 | cut -c1
+  fi
 
   while true; do
-    read default
-    case \${default} in
+    if [[ -z \"\${1}\" ]]; then
+      read default
+    else
+      default=\"\${1}\"
+      if [[ \"\${default}\" != 'b' ]] && [[ \"\${default}\" != 'g' ]] && [[ \"\${default}\" != 'l' ]]; then
```

Comment:
See if this works:
```
if ! [[ \"\${default}\" =~ "^[b|g|l]$" ]]; then

Reply:
No. It doesn't.
```
chronos@localhost ~/Download/source/chromebrew $ ld_default b
l
Invalid linker configuration: b
chronos@localhost ~/Download/source/chromebrew $ ld_default g
l
Invalid linker configuration: g
chronos@localhost ~/Download/source/chromebrew $ ld_default l
```

Reply:
You might need to escape more characters.  I'm thinking `|` is one of them.

Reply:
Anyway, not worth wrestling with it unless you want to do it.

Reply:
I don't think `|` is a special character in Ruby. I'd have to check, though.